components/common/src/templating/config.rs

```diff
@@ -588,34 +565,156 @@ fn is_toml_value_a_table(key: &str, table: &toml::value::Table) -> bool {
 }
 
 #[cfg(not(windows))]
-fn set_permissions<T: AsRef<Path>>(path: T, pkg: &Pkg) -> hcore::error::Result<()> {
+fn set_permissions<T: AsRef<Path>>(path: T, user: &str, group: &str) -> hcore::error::Result<()> {
     use crate::hcore::os::users;
     use crate::hcore::util::posix_perm;
 
     if users::can_run_services_as_svc_user() {
-        posix_perm::set_owner(path.as_ref(), &pkg.svc_user, &pkg.svc_group)?;
+        posix_perm::set_owner(path.as_ref(), &user, &group)?;
+    }
+    if path.as_ref().is_dir() {
+        posix_perm::set_permissions(path.as_ref(), CONFIG_DIR_PERMISSIONS)
+    } else {
+        posix_perm::set_permissions(path.as_ref(), CONFIG_PERMISSIONS)
```

Comment:
Since the activity here is the same, but the argument is different, could we have the `if`/`else` block determine the permissions to use, and then we pass that into a single `set_permissions` call?

Reply:
+1.  That's what I was looking for. Thanks!